nipype/interfaces/mrtrix3/preprocess.py

```diff
@@ -11,6 +11,78 @@
 from .base import MRTrix3BaseInputSpec, MRTrix3Base
 
 
+class DWIDenoiseInputSpec(MRTrix3BaseInputSpec):
+    in_file = File(
+        exists=True,
+        argstr='%s',
+        position=-2,
+        mandatory=True,
+        desc='input DWI image')
+    mask = File(
+        exists=True,
+        argstr='-mask %s',
+        position=1,
+        mandatory=False,
+        desc='mask image')
+    extent = traits.Tuple((traits.Int, traits.Int, traits.Int),
+        argstr='-extent %d,%d,%d',
+        mandatory=False,
+        desc='set the window size of the denoising filter. (default = 5,5,5)')
+    noise = File(
+        argstr='-noise %s',
+        mandatory=False,
+        desc='noise map')
+    out_file = File('dwi_denoised.mif',
+        exists=False,
+        usedefault=True,
+        argstr="%s",
+        position=-1,
+        desc="the output denoised DWI image")
```

Comment:
Consider using the following:

```Python
out_file = File(
    name_template='%s_denoised',
    name_source='in_file', 
    keep_extension=True,
    argstr="%s",
    position=-1,
    desc="the output denoised DWI image")
```

This will also auto-fill `out_file` in the output spec, so you can drop `_list_outputs`.